portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,104 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.CommentHistory;
+import java.util.ArrayList;
+import java.util.List;
+
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    //private List<CommentHistory> history = new ArrayList<>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("CommentHistory").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<CommentHistory> list = new ArrayList<CommentHistory>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
+      String comment = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      CommentHistory c = new CommentHistory(id, name,comment, timestamp);
+      list.add(c);
+    }
+
+  
+  
+    response.setContentType("application/json");
+    String json = new Gson().toJson(list);
+    response.getWriter().println(json);
+    
+    
+    
+
+    
+}
+private String convertToJson() {
+    String json = "{";
+    json += "\"acc\"";
+    json += ", ";
+    json += "\"acc\"";
+    json += "}";
+    return json;  
+}
+
+private String convertToJsonUsingGson(PresonIn p1) {
+    Gson gson = new Gson();
+    String json = gson.toJson(p1);
+    return json;
+  }
+@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+   // CommentHistory com=new CommentHistory();
+    String text = getParameter(request, "name-input", "");
+   
+    String text1 = getParameter(request, "comment-input", "");
+    //set 
+    // com.setName(text);
```

Comment:
Same note here as above - you can just delete the lines you don't need instead of commenting them out.